### ISSUE TITLE
Disallow addition of duplicate `location_name` entries

### ISSUE THREAD

**Issue by ehwenk** [traitecoevo/traits.build#46]
If you read in a locations table, where, there happen to be multiple rows for a given `location_name`, you get the following output in the metadata.yml file:

```
locations:
Atherton:
    description:
    - Tropical rain forest vegetation
    - Tropical rain forest vegetation
    - Tropical rain forest vegetation
    latitude (deg):
    - -17.117
    - -17.117
    - -17.117
    longitude (deg):
    - 145.65
    - 145.65
    - 145.65
  Cape Tribulation:
    description:
    - Complex mesophyll vine forest in tropical rain forest
    - Complex mesophyll vine forest in tropical rain forest
    - Complex mesophyll vine forest in tropical rain forest
etc.
``

At least there should be an error, indicating this duplication.

But, actually it would be great if the `metadata_add_locations()` function could simply incorporate a `distinct()` to remove the duplication. This would require one fewer tweaks by the data-input person - and would make it easy to simply add locations and location_properties directly from the data.csv file (for the many studies where location name, latitude, longitude, etc are columns in the data.csv file)

I'd initially thought it would be as simple as adding a `distinct()` after the `select` in the following code (setup.R, line 363), but of course this doesn't include `location_name` and there are many studies where that would be required as part of the "distinct". Maybe after the split?
```
  metadata$locations <- location_data %>%
    dplyr::select(dplyr::all_of(keep)) %>%
    split(location_data[[location_name]]) %>%
    lapply(as.list)
```

It might be good to have a test/error that triggers if there are multiple rows with the same `location_name` remaining after running `distinct()`?

**Comment by ehwenk** [comment]
The simple solution to this would be to add ```location_data %>% distinct(location_data[[location_name]])``` to line 365 of setup.R. However there are two other things to consider:
- There should be a warning added if nrow changes during this
- If there are multiple rows of data for a given location name, the function should break if the location properties aren't identical across all rows; sometimes the same location name is used for different "locations"

**Comment by ehwenk** [comment]
closed with [97d4593](https://github.com/traitecoevo/traits.build/commit/97d459317f6b215a0086b3d9e8bbcd66f6eb1a25)
